### PR TITLE
[Feature 22 23] User needs to upload an image using the camera or upload the document from the device.

### PR DESCRIPTION
**Pull request by alexxispn**
## Description

This pull request adds a new feature for the register process, the "RegisterReferral" page, which allows the user to upload a file (jpg/png/jpeg/pdf) and take a photo using their device's camera.

## Changes

- Add a new color - "quarter-color" from Figma.
- Add i18n text for the RegisterReferral page.
- Add cross (remove) icon and uploadImage icon.
- Update AppNextButton to accept an additional "onClick" prop.
- Add a new page, "RegisterReferral," to the Router.
- Add navigation for the "RegisterReferral" page to the "RegisterId" page.
- Create the "RegisterReferral" page with a custom form and custom hook.

## Checks

- [X] Project Builds
- [X] Project passes tests and checks
- [x] Updated documentation accordingly
- [x] "Enviar solicitud" button is disabled until a document is attached.
- [x] Document types could be PDF, PNG, JPG or JPEG.
- [x] It cannot exceed 5 mb
- [x] Screen follows provided designs

## Additional information

- Issue #7 

This branch was created from the branch 'feature-24/build-thank-you-page-after-onboarding-process' in order not to have to redo tasks already done.
URL: https://github.com/TheTributeCommunity/fesbal-frontend/pull/54

### CODE REVIEW

File: src/components/atom/AppNextButton.tsx [hunk 1]
```diff
@@ -23,7 +23,10 @@ const AppNextButton: FC<ButtonProps> = ({title, disabled = false, bgColor = "bg-
     );
     return (
         <button
-            className={btnClasses} disabled={disabled} type="submit">
+            className={btnClasses} disabled={disabled} {...(onClick
```
Comment (yodra) [inline src/components/atom/AppNextButton.tsx:26]:
```
{...(onClick ? {type: "button", onClick} : {type: "submit"})}
```

What do you think if we extract this in a variable to have the component clean?

My suggestion in this case is:
```
let buttonType = onClick ? {type: "button", onClick} : {type: "submit"};
    return (
        <button className={btnClasses} disabled={disabled} {...buttonType}>
            {title}
        </button>
    );
```

File: src/components/molecules/RegisterReferralForm.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                                {t("upload")}
+                            </button>
+                            <input type="file" id="camera" accept="image/*" capture onChange={handleFileChange}
+                                   style={{display: "none"}} ref={cameraRef}/>
+                            <button className="mt-4 rounded-2xl px-2 py-5 text-white font-button bg-primary-color"
+                                    onClick={() => handleClick(cameraRef)}>
+                                {t("camera")}
+                            </button>
+                        </div>
+                    </div>}
```
Comment (yodra) [inline src/components/molecules/RegisterReferralForm.tsx:34]: I usually divide this type of component into different components.
If it doesn't have a file, one component is needed, and another when it has a file

**Review by yodra — COMMENTED** [review]
Good job! Please check the comments and if you have any question let me know 😉

File: src/components/molecules/RegisterReferralForm.tsx (hunk 2 above)
Comment (alexxispn) [reply src/components/molecules/RegisterReferralForm.tsx]: Hi @yodra! I have completed the changes on this pull request as requested. I have refactored the RegisterReferralForm component into smaller components and refactored the AppNextButton to add a buttonType on it. Please let me know if there is anything else that needs to be implemented or if there are any improvements that can be made to the current changes. Thank you! 😄

**Review by rdiaz82 — APPROVED** [review]
LGTM

**Pull request merged**